app/fields/date_field.rb

```diff
@@ -35,4 +35,14 @@ def self.from_persisted_value(date)
     end
   end
 
+  def set_error_message_lookup_proc!(error_proc)
+    @error_message_lookup_proc = error_proc
+  end
+
+  private
+
+  def error_message_lookup_proc
+    @error_message_lookup_proc || ->(a){ a }
```

Comment:
Do you want to cache the lookup proc with `||=`? Or is it on purpose always creating a new one.